data/transactions/logicsig.go

```diff
@@ -54,22 +54,35 @@ func (lsig *LogicSig) Len() int {
 	return lsiglen
 }
 
-// Equal returns true if both LogicSig are equivalent
+// Equal returns true if both LogicSig are equivalent.
+//
+// Out of paranoia, Equal distinguishes zero-length byte slices
+// from byte slice-typed nil values as they may have subtly
+// different behaviors within the evaluation of a LogicSig,
+// due to differences in msgpack encoding behavior.
 func (lsig *LogicSig) Equal(b *LogicSig) bool {
```

Comment:
seems like a test that captures this behavior (programs are the same for escrow accounts) is needed